module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -165,33 +160,133 @@ namespace module::input {
-            mean.Rwt           = this->config.motionFilter.initial.mean.rotation;
-            mean.omegaTTt      = this->config.motionFilter.initial.mean.rotationalVelocity;
-            mean.omegaTTt_bias = this->config.motionFilter.initial.mean.gyroscopeBias;
+            mean.rTWw     = this->config.motionFilter.initial.mean.position;
+            mean.vTw      = this->config.motionFilter.initial.mean.velocity;
+            mean.Rwt      = this->config.motionFilter.initial.mean.rotation;
+            mean.omegaTTt = this->config.motionFilter.initial.mean.rotationalVelocity;
 
             MotionModel<double>::StateVec covariance;
-            covariance.rTWw          = this->config.motionFilter.initial.covariance.position;
-            covariance.vTw           = this->config.motionFilter.initial.covariance.velocity;
-            covariance.Rwt           = this->config.motionFilter.initial.covariance.rotation;
-            covariance.omegaTTt      = this->config.motionFilter.initial.covariance.rotationalVelocity;
-            covariance.omegaTTt_bias = this->config.motionFilter.initial.covariance.gyroscopeBias;
+            covariance.rTWw     = this->config.motionFilter.initial.covariance.position;
+            covariance.vTw      = this->config.motionFilter.initial.covariance.velocity;
+            covariance.Rwt      = this->config.motionFilter.initial.covariance.rotation;
+            covariance.omegaTTt = this->config.motionFilter.initial.covariance.rotationalVelocity;
             motionFilter.set_state(mean.getStateVec(), covariance.asDiagonal());
+
+            // Don't filter any sensors until we have initialised the filter
+            update_loop.disable();
+            reset_filter.store(true);
         });
 
         on<Configuration>("FootDownNetwork.yaml").then([this](const Configuration& config) {
             // Foot load sensor config
             load_sensor = VirtualLoadSensor<float>(config);
         });
 
-        on<Last<20, Trigger<RawSensors>>, Single>().then(
-            [this](const std::list<std::shared_ptr<const RawSensors>>& sensors) {
+        on<Last<20, Trigger<RawSensors>>, With<KinematicsModel>, Single>().then(
+            [this](const std::list<std::shared_ptr<const RawSensors>>& sensors, const KinematicsModel& model) {
+                // If we need to reset the filter, do that here
+                if (reset_filter.load()) {
+                    Eigen::Vector3d acc   = Eigen::Vector3d::Zero();
+                    Eigen::Vector3d gyro  = Eigen::Vector3d::Zero();
+                    Eigen::Vector3d rMFt  = Eigen::Vector3d::Zero();
+                    auto filtered_sensors = std::make_unique<Sensors>();
+
+                    for (const auto& s : sensors) {
+                        // Accumulate accelerometer and gyroscope readings
+                        acc += Eigen::Vector3d(s->accelerometer.x, s->accelerometer.y, s->accelerometer.z);
```

Comment:
Here's a spicy suggestion, which probably stinks, but it was fun to write lol.
Instead of doing the accumulation inside the loop, wb this
```suggestion
                        acc = std::accumulate(sensors.begin(), sensors.end(), acc, [](Eigen::Vector3d sum, auto s) {return sum + Eigen::Vector3d(s->accelerometer.x, s->accelerometer.y, s->accelerometer.z) ; }) / sensors.size();
```

_it's IdIoMaTiC!!1!_

Reply:
We are using the loop for other purposes anyway, I dont think the idiom extends this far

Reply:
fair enough